base/sort.jl

```diff
@@ -373,12 +382,12 @@ julia> searchsortedlast([1, 2, 4, 5, 5, 7], 0) # no match, insert at start
 """ searchsortedlast
 
 """
-    insorted(x, a; by=<transform>, lt=<comparison>, rev=false) -> Bool
+    insorted(x, a; by=identity, lt=isless, rev=false) -> Bool
 
 Determine whether an item `x` is in the sorted collection `a`, in the sense that
-it is [`==`](@ref) to one of the values of the collection according to the order
-specified by the `by`, `lt` and `rev` keywords, assuming that `a` is already
-sorted in that order, see [`sort`](@ref) for the keywords.
+it is [`==`](@ref) to one of the values of the collection. The `by`, `lt` and
+`rev` keywords modify what order is assumed for the collection, as described in
+the [`sort!`](@ref) documentation.
```

Comment:
This docstring is still inconsistent with behavior as reported by https://github.com/JuliaLang/julia/issues/44102. `==` is unused and misleading.

Reply:
Indeed. In the latest commit I've made substantive changes to the `searchsortedXXX` and `insorted` docstrings, I think the behaviors are correctly described now.

The new docstrings are inspired by the documentation of the C++ standard library: [equal_range](https://cplusplus.com/reference/algorithm/equal_range/)  (equivalent of our `searchsorted`) and [binary_search](https://cplusplus.com/reference/algorithm/binary_search/) (equivalent of our `insorted`). In particular, I've added to the docstrings that these function also work for vectors that are not sorted but correctly partitioned with respect to the searched value.